spark/v3.5/spark/src/main/java/org/apache/iceberg/spark/actions/SparkShufflingDataRewriter.java

```diff
@@ -86,6 +88,16 @@ protected SparkShufflingDataRewriter(SparkSession spark, Table table) {
 
   protected abstract org.apache.iceberg.SortOrder sortOrder();
 
+  /**
+   * Retrieves and returns the schema for the rewrite using the current table schema.
```

Comment:
why do we need a special method for "sortSchema" here? Shouldn't it just be schema()?

Reply:
I think, we need to keep this method, `SparkZOrderDataRewriter` `sortSchema()` need all Z_SCHEMA columns added explicitly.  `org.apache.iceberg.spark.actions.SparkShufflingDataRewriter#outputSortOrder` is common between `SparkZOrderDataRewriter`  and `SparkBinPackDataRewriter`